internal/bufferpool/buffer.go

```diff
@@ -164,3 +164,22 @@ func overwriteData(bs []byte) {
 		bs[i] = byte(i)
 	}
 }
+
+// AutoReleaseBuffer wraps a Buffer in a io.ReadCloser implementation
+// that returns the underlying Buffer to the pool on Close().
+type AutoReleaseBuffer struct {
```

Comment:
We could avoid creating a wrapper buffer object by exposing Close method on the existing buffer object, and closing buffer puts the buffer object back into the pool similar to existing Release method. Thoughts?

Reply:
I don't have a strong opinion either way.

Merging them into a single buffer object avoids an allocation. But semantically, does it make sense? I'm not so sure. The separate wrapper object allows us to extend the functionality of Buffer for a specific use-case that may not be generally useful in the underlying Buffer object.